### DISCUSSION TITLE
How to get user drawings in foliumap?

### DISCUSSION THREAD

**Discussion by swdmike** [giswqs/streamlit-geospatial#119] [Q&A] [answered]
import leafmap.foliumap as foliumap
map = foliumap.Map(location=[30, 110], zoom_start=9)

I want to get user drawing in the map, so I use
roi = map.user_roi_bounds()
but it prints 'The folium plotting backend does not support this function'.

Then how do I get the user drawings using foliumap?

Thanks.

**Comment by giswqs** [comment, marked as answer]
folium does not support getting drawings programmatically. See https://github.com/python-visualization/folium/issues/1565 and https://github.com/python-visualization/folium/issues/1402

You need ipyleaflet.

**Reply by swdmike** [reply]
I am doing a streamlit app, is it possible to use ipyleaflet?

**Reply by giswqs** [reply]
No. Streamlit does not support ipyleaflet. See https://github.com/opengeos/streamlit-geospatial/blob/master/pages/1_%F0%9F%93%B7_Timelapse.py#L259
